wire-runtime/src/main/java/com/squareup/wire/ExtensionRegistry.java

```diff
@@ -42,16 +43,8 @@
   }
 
   @SuppressWarnings("unchecked")
-  public <T extends ExtendableMessage<T>, E> Extension<T, E>
-      getExtension(Class<T> messageClass, int tag) {
-    Map<Integer, Extension<?, ?>> map = extensionsByTag.get(messageClass);
-    return map == null ? null : (Extension<T, E>) map.get(tag);
-  }
-
-  @SuppressWarnings("unchecked")
-  public <T extends ExtendableMessage<T>, E> Extension<T, E>
-      getExtension(Class<T> messageClass, String name) {
-    Map<String, Extension<?, ?>> nameMap = extensionsByName.get(messageClass);
```

Comment:
I think you can kill `extensionsByName` map and associated population in the loop.